apps/explorer/src/components/TransactionBlocks/TransactionBlocks.tsx

```diff
@@ -0,0 +1,152 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { type PaginatedTransactionResponse } from '@mysten/sui.js';
+import { type InfiniteData } from '@tanstack/react-query';
+import { useState } from 'react';
+
+import { genTableDataFromTxData } from '../transactions/TxCardUtils';
+
+import {
+    DEFAULT_TRANSACTIONS_LIMIT,
+    useGetTransactionBlocks,
+} from '~/hooks/useGetTransactionBlocks';
+import { Heading } from '~/ui/Heading';
+import { Pagination } from '~/ui/Pagination';
+import { PlaceholderTable } from '~/ui/PlaceholderTable';
+import { RadioGroup, RadioOption } from '~/ui/Radio';
+import { TableCard } from '~/ui/TableCard';
+
+type CurrentPageFilter = {
+    from: number;
+    to: number;
+};
+
+enum TRANSACTION_FILTERS {
+    FROM = 'from',
+    TO = 'to',
+}
+
+type TransactionBlocksProps = {
+    address: string;
+};
+
+function TransactionBlocks({ address }: TransactionBlocksProps) {
+    const [filterValue, setFilterValue] = useState<TRANSACTION_FILTERS>(
+        TRANSACTION_FILTERS.FROM
+    );
+    const [currentPage, setCurrentPage] = useState<CurrentPageFilter>({
+        from: 0,
+        to: 0,
+    });
+    const {
+        data,
+        isLoading,
+        isFetching,
+        isFetchingNextPage,
+        fetchNextPage,
+        hasNextPage,
+    } = useGetTransactionBlocks(
+        address,
+        filterValue === TRANSACTION_FILTERS.FROM
+    );
+
+    const setFilter = (value: TRANSACTION_FILTERS) => {
+        setFilterValue(value);
+    };
+
+    const generateTableCard = (
+        currentPage: CurrentPageFilter,
+        filterValue: TRANSACTION_FILTERS,
+        data?: InfiniteData<PaginatedTransactionResponse>
+    ) => {
+        if (!data) {
+            return;
+        }
+        const cardData = genTableDataFromTxData(
+            data?.pages[currentPage[filterValue]].data
+        );
+        return <TableCard data={cardData.data} columns={cardData.columns} />;
+    };
+
+    return (
+        <div>
+            <div className="flex items-center justify-between border-b border-gray-45 pb-5">
+                <Heading color="gray-90" variant="heading4/semibold">
+                    Transaction Blocks
+                </Heading>
+                <RadioGroup
+                    className="flex"
+                    ariaLabel="transaction filter"
+                    value={filterValue}
+                    onChange={setFilter}
```

Comment:
You can pass `setFilterValue` directly here and remove the need for `setFilter` function on line 54

Reply:
ooh nice catch

Reply:
You can also change the `TRANSACTION_FILTERS` into an array of object with `label` and `value` and just map them too on the `RadioOption` so if we need to expand it, it will all be in 1 place.

Reply:
I like that but then I sort of lose the maintainability of the CurrentPageState is the pitfall there

Reply:
O ya I see. That is fine then.